stock_dynamic_routing/models/stock_move.py

```diff
@@ -236,13 +237,27 @@ def _apply_routing_rule_pull(self, routing_details):
         """
         pickings_to_check_for_emptiness = self.env["stock.picking"]
         move_ids_to_assign_per_location = defaultdict(list)
-        for move in self:
-            routing_rule = routing_details[move].rule
-            if not routing_rule.method == "pull":
+        move_ids_to_assign_nonrelocated = []
+        next_moves_to_update = self.browse()
+        routing_to_apply = [
+            (move, detail.rule) for move, detail in routing_details.items()
+        ]
+        for move, routing_rule in routing_to_apply:
+            if routing_rule.method == "push":
```

Comment:
The `method` selection field has no default value nor is required, so we could have no value here
```suggestion
            if routing_rule.method != "pull":
```

Reply:
Or having no value means it should be applied on both push & pull?